src/object.cpp

```diff
@@ -337,4 +338,89 @@ extern "C"
 
         ddwaf_object_invalid(object);
     }
+
+    DDWAF_OBJ_TYPE ddwaf_object_type(ddwaf_object* object)
+    {
+        return object ? object->type : DDWAF_OBJ_INVALID;
+    }
+
+    size_t ddwaf_object_size(ddwaf_object* object)
+    {
+        if (object == NULL || !IS_CONTAINER(object))
+        {
+            return 0;
+        }
+
+        return object->nbEntries;
+    }
+
+    size_t ddwaf_object_length(ddwaf_object* object)
+    {
+        if (object == NULL || object->type != DDWAF_OBJ_STRING)
+        {
+            return 0;
+        }
+
+        return object->nbEntries;
+    }
+
+    const char* ddwaf_object_get_key(ddwaf_object* object, size_t* length)
+    {
+        if (object == NULL || object->parameterName == NULL)
+        {
+            return NULL;
+        }
+
+        if (length)
```

Comment:
Should the length parameter be optional?
We have no guarantee that strings are NULL-terminated, do we?

Reply:
We absolutely do have a guarantee for `ddwaf_object` returned by the WAF. The same is not necessarily applicable to input objects, depending on how they are created, but this API is not meant for that purpose.